### DISCUSSION TITLE
Setting up system clipboard with * and + registers?

### DISCUSSION THREAD

**Discussion by volkswagenfeature** [nix-community/nixvim#81] [Q&A] [answered]
Curious how you all are enabling system clipboard access. I started by installing `xclip`, then setting the following in my configs:
```
programs.nixvim.options.clipboard = {
          name = "Testclip";
          copy = {
            "+" = ["xclip"];
            "*" = ["xclip"];
          };
          paste = {
            "+" = ["xclip"];
            "*" = ["xclip"];
          };
          cache_enabled = true;
}
```
This is basically just what they suggest in `:help clipboard`. It doesn't work, so I'm curious how you guys have done it.

**Comment by pta2002** [comment, marked as answer]
Hey, it looks like what you want to set is `g:clipboard`, and not the clipboard option (options are set with `:set` in vimscript, e.g. `opts.numbers = true` is the same as `:set numbers`)

You can set this global setting `globals.clipboard` instead. I'd also suggest adding `xclip` by adding it to Nixvim's `extraPackages`: `programs.nixvim.extraPackages = [ pkgs.xorg.xclip ];`.

If you want to be able to copy to the system clipboard without specifying the `+` register, just set the `clipboard` option (not global!) to `unnamedplus`:

`programs.nixvim.opts.clipboard = [ "unnamedplus" ];`.

Hope this helps! Let me know if you need anything else.

**Reply by volkswagenfeature** [reply]
Thank you very much for the clarification! You used `options` in the docs, but I missed `globals` all the way at the bottom. Is there a way to see where these are defined? I've recently discovered `nix repl`. Some other flakes list options in `nixosConfigurations.[hostname].config.[package]` or other similar locations... But I'm unsure how to see what features have been written for the flake so far: `nixosConfigurations.hostname.config.nixvim` only contains `helpers`, which contains a lot of stuff, none of which looks like options I can use in my nix file.

**Reply by pta2002** [reply]
There are module docs over on https://nix-community.github.io/nixvim.

I really need to fix those docs up because the auto generation hasn't been working for a few months, so there are a few plugins missing from there.

Still, it's probably enough for what you need!
